typescript/api-websocket-lambda-dynamodb/index.ts

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env node
+import {AssetCode, Function, Runtime} from "@aws-cdk/aws-lambda";
+import {CfnApi, CfnDeployment, CfnIntegration, CfnRoute, CfnStage} from "@aws-cdk/aws-apigatewayv2";
+import {App, ConcreteDependable, Construct, Duration, RemovalPolicy, Stack, StackProps} from '@aws-cdk/core';
+import {Effect, ManagedPolicy, PolicyStatement, Role, ServicePrincipal } from "@aws-cdk/aws-iam";
+import {AttributeType, Table} from "@aws-cdk/aws-dynamodb";
+
+import config from './config.json';
+
+class ChatAppStack extends Stack {
+    constructor(scope: Construct, id: string, props?: StackProps) {
+        super(scope, id, props);
+        const tableName = "simplechat_connections";
+
+        // initialise api
+        const name = id + "-api"
+        const api = new CfnApi(this, name, {
+            name: "ChatAppApi",
+            protocolType: "WEBSOCKET",
+            routeSelectionExpression: "$request.body.action",
+        });
+        const table = new Table(this, `${name}-table`, {
+            tableName: tableName,
+            partitionKey: {
+                name: "connectionId",
+                type: AttributeType.STRING,
+            },
+            readCapacity: 5,
+            writeCapacity: 5,
+            removalPolicy: RemovalPolicy.DESTROY
+        });
+
+        // initialise lambda and permissions
+
+        const lambdaPolicy = new PolicyStatement({
+            actions: [
+                "dynamodb:GetItem",
+                "dynamodb:DeleteItem",
+                "dynamodb:PutItem",
+                "dynamodb:Scan",
+                "dynamodb:Query",
+                "dynamodb:UpdateItem",
+                "dynamodb:BatchWriteItem",
+                "dynamodb:BatchGetItem",
+                "dynamodb:DescribeTable",
+                "dynamodb:ConditionCheckItem"
+            ],
+            resources: [table.tableArn]
+        });
+
+        const connectLambdaRole = new Role(this, "connectLambdaRole", {
```

Comment:
`connectLambdaRole` is used only in one lambda. Maybe will be reasonable just to link them with `grant` like:
`connectFunc.grantReadWrite(table.tableArn)`

Reply:
I have updated it.